src/grep.rs

```diff
@@ -11,7 +11,7 @@ use crate::args::{self, Args};
 /// Build the regex pattern with the given options.
 /// By default, the `unicode` flag is set to false, and `dot_matches_new_line` set to true.
 fn build_pattern(
-  pattern: &String,
+  pattern: &str,
```

Comment:
This part is already adressed in #1 , so I'll have it from there. Can you revert this part please?